Bento/Decorators/CustomInputComponent.swift

```diff
@@ -9,37 +10,66 @@ extension Renderable {
         return CustomInputComponent(
             source: self,
             customInput: input,
-            contentStatus: contentStatus,
-            highlightColor: highlightColor
+            focusEligibility: .eligible(contentStatus),
+            highlightColor: highlightColor,
+            focusesOnFirstDisplay: false
+        ).asAnyRenderable()
+    }
+
+    /// Sets up `customInput` to prepare for presention when user taps `self`,
+    /// and also presents it immediately when `state` is `.some`.
+    ///
+    /// - important: This method is useful when `customInput` needs to be displayed asynchronously after state change.
+    /// - note: Due to asynchronous presentation, `focusEligibility` is not supported.
+    public func customInputImmediately<State>(
+        when state: State?,
+        input: (State) -> CustomInput,
+        highlightColor: UIColor? = UIColor(red: 239/255.0, green: 239/255.0, blue: 244/255.0, alpha: 1)
+    ) -> AnyRenderable {
+        return CustomInputComponent(
```

Comment:
Do we really need to change `CustomInputComponent` to support this? Why can't we just return `self` when state is `nil`? At least I think this should be done the same way that we build boxes where sections and/or nodes are conditionally composed depending on a certain condition and here I think we could follow the same principle, if the `state` is `nil` it means there's no `CustomInput` so we don't need a `CustomInputComponent` but once that changes we can then build a `CustomInputComponent` to display that input.

I'm not sure if I'm missing some detail.

Reply:
`state = nil` doesn't mean we don't use `customInput` at all.
It's rather the same behavior as we already have: "Show customInput when user tapped"

On the contrary, when `state` exists, `customInput` shows **immediately** when re-rendered.
It also holds the same property  "show customInput when user tapped" as well.

Reply:
Maybe I should rename method e.g. `func customInput(immediatelyWhen:...)` so that "immediately" belongs to state existence.

Reply:
> state = nil doesn't mean we don't use customInput at all.
It's rather the same behavior as we already have: "Show customInput when user tapped"

I'm a bit confused, in your example you tap `Gender` and there's no input being presented, you display an activity indicator as accessory and after 1 sec. your state changes and then it shows the option picker, why do we need to have a custom input before the change of state?

Reply:
> why do we need to have a custom input before the change of state

That's good question!
IIRC I tried the approach by removing `customInput` only when loading.
But this caused Bento diffing animation for the entire "gender cell" that I wanted to avoid.
So, keeping `customInput` throughout the handling is (probably) required to avoid such animations.

Reply:
Oh yes it does that because we are swapping from one component to another one if we do that 😕

I don't see any way to avoid that animation but maybe @andersio has an idea.

Reply:
There could be some diffing optimization, but I think it's still OK to always keep `customInput` during loading.
It might still be able to tap though (which will be a bug), I can add `isEnabled` property to prevent that, so that we don't need to worry about removal animation of `customInput`.

Reply:
(For sure diffing optimization will be more nice, but we won't probably invest too much here)

Reply:
#146 is meant to tackle that.